### PR TITLE
story: finalize extra dashboard widgets 

### PR DESCRIPTION
**Pull request by mbritense**
closes https://ritense.tpondemand.com/entity/114293-febe-finalize-dashboard-widgets

### CODE REVIEW

File: projects/valtimo/dashboard/src/lib/data-sources/case-counts/components/case-counts-configuration/case-counts-configuration.component.html [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  <cds-label class="conditions__label">{{
+    'counts' | widgetTranslate: dataSourceKey | async
+  }}</cds-label>
+
+  <p class="conditions__helper-text cds--form__helper-text">
+    {{ 'countsHelperText' | widgetTranslate: dataSourceKey | async }}
+  </p>
+
+  <cds-tile
+    *ngFor="let _ of queryItemsList$ | async; let i = index"
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/data-sources/case-counts/components/case-counts-configuration/case-counts-configuration.component.html]: Use @for

File: projects/valtimo/dashboard/src/lib/display-types/bar-chart/components/bar-chart-display/bar-chart-display.component.ts [hunk 2]
```diff
@@ -25,7 +27,44 @@ import {BarChartData, BarChartDisplayTypeProperties} from '../../models';
...
+  public readonly barChartData$: Observable<ChartTabularData> = this._data$.pipe(
+    filter(data => !!data),
+    map(
+      data =>
+        data?.values.map(dataValue => ({
+          group: dataValue.label,
+          value: dataValue.value,
+        })) || []
+    ),
+    tap(data => console.log(data))
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/bar-chart/components/bar-chart-display/bar-chart-display.component.ts]: Remove this

File: projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-configuration/donut-configuration.component.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  @Input() displayTypeKey: string;
+  @Input() set disabled(disabledValue: boolean) {
+    if (disabledValue) {
+      this.form.disable();
+    } else {
+      this.form.enable();
+    }
+  }
+
+  public get title() {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-configuration/donut-configuration.component.ts]: return type

File: projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-configuration/donut-configuration.component.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    return this.form.get('title');
+  }
+
+  public get subtitle() {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-configuration/donut-configuration.component.ts]: return type

File: projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-configuration/donut-configuration.component.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    return this.form.get('subtitle');
+  }
+
+  public get label() {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-configuration/donut-configuration.component.ts]: return type

File: projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-display/donut-display.component.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+<section class="valtimo-donut-widget__header">
+  <h3 class="valtimo-donut-widget__title">{{ displayTypeProperties.title }}</h3>
+
+  <span *ngIf="displayTypeProperties.subtitle" class="valtimo-donut-widget__subtitle">
+    {{ displayTypeProperties.subtitle }}
+  </span>
+</section>
+
+<section class="valtimo-donut-widget__content">
+  <ng-container *ngIf="donutData$ | async as donutData">
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-display/donut-display.component.html:26]: Can this async be added directy to the if?

File: projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-display/donut-display.component.scss [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+ * limitations under the License.
+ */
+
+@use '@carbon/colors';
+
+:host {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  width: 100%;
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-display/donut-display.component.scss:23]: width: 100% may not be necessary on flex-box elements. Not sure tho

File: projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-display/donut-display.component.scss [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
+
+  &__subtitle {
+    font-size: 14px;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/donut/components/donut-display/donut-display.component.scss]:
```suggestion
  &__title,
  &__subtitle {
     white-space: nowrap;
     overflow: hidden;
     text-overflow: ellipsis;
  }
  
  &__title {
    font-size: 20px;
    margin-bottom: 12px;
  }

  &__subtitle {
    font-size: 14px;
  }
```

File: projects/valtimo/dashboard/src/lib/display-types/gauge/components/gauge-configuration/gauge-configuration.component.ts [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  @Input() displayTypeKey: string;
+  @Input() set disabled(disabledValue: boolean) {
+    if (disabledValue) {
+      this.form.disable();
+    } else {
+      this.form.enable();
+    }
+  }
+
+  public get title() {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/gauge/components/gauge-configuration/gauge-configuration.component.ts]: return type

File: projects/valtimo/dashboard/src/lib/display-types/gauge/components/gauge-configuration/gauge-configuration.component.ts [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    return this.form.get('title');
+  }
+
+  public get subtitle() {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/gauge/components/gauge-configuration/gauge-configuration.component.ts]: return type

File: projects/valtimo/dashboard/src/lib/display-types/gauge/components/gauge-configuration/gauge-configuration.component.ts [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    return this.form.get('subtitle');
+  }
+
+  public get label() {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/gauge/components/gauge-configuration/gauge-configuration.component.ts]: return type

File: projects/valtimo/dashboard/src/lib/display-types/gauge/components/gauge-display/gauge-display.component.scss [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  width: 100%;
+  height: 100%;
+  padding: 20px;
+}
+
+.valtimo-gauge-widget {
+  &__title {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/gauge/components/gauge-display/gauge-display.component.scss]: Same type of comment as above, group the common properties of title and subtitle

File: projects/valtimo/dashboard/src/lib/display-types/meter/components/meter-display/meter-display.component.scss [hunk 13]
```diff
@@ -14,9 +14,43 @@
...
-  padding: 20px;
+  flex-direction: column;
+  justify-content: space-between;
   width: 100%;
   height: 100%;
+  padding: 20px;
+}
+
+.valtimo-donut-widget {
+  &__title {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/display-types/meter/components/meter-display/meter-display.component.scss]: Same thing as before

File: projects/valtimo/dashboard/src/lib/models/configuration.model.ts [hunk 14]
```diff
@@ -16,9 +16,9 @@
 
 import {EventEmitter} from '@angular/core';
 
-interface ConfigurationOutput {
+interface ConfigurationOutput<T = object> {
```
Comment (teodor-ritense) [inline projects/valtimo/dashboard/src/lib/models/configuration.model.ts:19]: Do you need the '= object'?

Comment (mbritense) [reply projects/valtimo/dashboard/src/lib/models/configuration.model.ts:19]: Yes, needs to default to something if type is not passed

**Review by teodor-ritense — APPROVED** [review]

**Pull request merged**